docs/sources/migration-guide/migrating-from-thanos-or-prometheus.md

```diff
@@ -78,15 +78,254 @@ If you need to have external labels in your query results, this is currently not
+   level=info ts=2022-10-10T13:05:28.675240198Z caller=tools_bucket.go:1231 msg="uploading new block" source=01GEGWPME2187SVFH63G8DH7KH new=01GF0ZWPWGEPHG5NV79NH9KMPV
+   level=info ts=2022-10-10T13:05:38.922348564Z caller=tools_bucket.go:1241 msg=uploaded source=01GEGWPME2187SVFH63G8DH7KH new=01GF0ZWPWGEPHG5NV79NH9KMPV
+   level=info ts=2022-10-10T13:05:38.979696873Z caller=block.go:203 msg="block has been marked for deletion" block=01GEGWPME2187SVFH63G8DH7KH
+   level=info ts=2022-10-10T13:05:38.979832767Z caller=tools_bucket.go:1249 msg="rewrite done" IDs=01GEGWPME2187SVFH63G8DH7KH
+   level=info ts=2022-10-10T13:05:38.980197796Z caller=main.go:161 msg=exiting
+   ```
+   
+   
+   Cleanup the original blocks which are marked for deletion.
+  
+   ```bash
+   thanos tools bucket cleanup \
+       --objstore.config-file bucket.yaml \
+       --delete-delay=0
+   ```
+  
+   > **Note**: If there are multiple prometheus clusters, then relabelling each of them in parallel may speed up the entire process.
+   > Get the list of blocks that for each cluster, and process it separately.
+   > ```bash
+   > thanos tools bucket inspect \
+   >     --objstore.config-file bucket.yaml \
+   >     --output=tsv \
+   >     | grep <PROMETHEUS-CLUSTER-NAME> \
+   >     | awk '{print $1}' > prod-blocks.tsv
+   >```
+  
+   >```bash
+   > for ID in `cat prod-blocks.tsv`
+   > do
+   >     thanos tools bucket rewrite  \
+   >        --objstore.config-file bucket.yaml \
+   >        --rewrite.to-relabel-config-file relabel-config.yaml \
+   >        --delete-blocks \
+   >        --no-dry-run \
+   >        --id $ID         
+   > done
+   >```
+ 
+5) Remove external labels from meta.json
+   
+   Mimir compactor will not be able to compact the blocks with external labels in their meta.json. Therefore these external labels have to be  removed before copying them to the Mimir bucket.
+   
+   Use the below script to remove the labels from the meta.json.
+   
+   ```bash
+   #!/bin/bash
+
+   BUCKET="GCS Bucket name"
+
+   echo "Fetching list of meta.json files (this can take a while if there are many blocks)"
+   gsutil ls "gs://$BUCKET/*/meta.json" > meta-files.txt
+
+   echo "Processing meta.json files"
+   for FILE in $(cat meta-files.txt); do
+      echo "Removing Thanos labels from $FILE"
+      ORIG_META_JSON=$(gsutil cat "$FILE")
+      UPDATED_META_JSON=$(echo "$ORIG_META_JSON" | jq "del(.thanos.labels)")
+
+      if ! diff -u <( echo "$ORIG_META_JSON" | jq . ) <( echo "$UPDATED_META_JSON" | jq .) > /dev/null; then
+         echo "Backing up $FILE to $FILE.orig"
+         gsutil cp "$FILE" "$FILE.orig"
+         echo "Uploading modified $FILE"
+         echo "$UPDATED_META_JSON" | gsutil cp - "$FILE"
+      else
+         echo "No diff for $FILE"
+      fi
+   done
+   ```
+   
+   Clean up the original meta.json file which were backed using the below command.
+   
+   ```bash
+   gsutil rm gs://<mimir-intermediate-bucket>/*/*.orig
+   ```
```

Comment:
This is not required. These files will be ignored by Mimir, and can be left around for troubleshooting, if needed. If we don't want them, we can as well not upload them in the previous script.